utils/supabase/middleware.ts

```
@@ -1,36 +1,51 @@
+import { cookies } from 'next/headers';
+import { NextResponse } from 'next/server';
+import { createClient } from './server';
 
-import { createServerClient, type CookieOptions } from "@supabase/ssr";
-import { type NextRequest, NextResponse } from "next/server";
+/**
+ * Middleware to handle authentication for API routes
+ * @param handler The route handler function
+ * @returns A new handler function with authentication
+ */
+export function withAuth(handler: Function) {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)